pages/cats.js

```diff
@@ -2,15 +2,27 @@ import React, { useState } from "react";
 import Link from "next/link";
 import List from "../components/Catlist/index.js";
 import cats from "../lib/cat_data.js";
+import Footer from "../components/Footer/index.js";
 
 function AnimalsPage() {
   const [filteredItems, setFilteredItems] = useState(cats);
   const [ageFilter, setAgeFilter] = useState("");
   const [genderFilter, setGenderFilter] = useState("");
-  const [speciesFilter, setSpeciesFilter] = useState("");
   const [locationFilter, setLocationFilter] = useState("");
   const [noResults, setNoResults] = useState(false);
 
+  const [favorites, setFavorites] = useState([]);
+
+  const handleBookmark = (id) => {
+    let newFavorites;
+    if (favorites.includes(id)) {
+      newFavorites = favorites.filter((favId) => favId !== id);
```

Comment:
like here, i now have an idea of there is sth. different inside `newFavorites` than inside `favorites`